docs: add description of returning different status codes (#10059)
nuxt/framework: docs/content/1.docs/2.guide/2.directory-structure/1.server.md

})
```

### Error handling

> If no errors are thrown a status code of `200 OK` will be returned. Any uncaught errors will return a `500 Internal Server Error` HTTP Error.

Reviewer: ```suggestion
If no errors are thrown, a status code of `200 OK` will be returned. Any uncaught errors will return a `500 Internal Server Error` HTTP Error.
```

To return other error codes, throw an exception with `createError`

```ts [server/api/validation/[id].ts]
export default defineEventHandler((event) => {
  const id = parseInt(event.context.params.id) as number
  if (!Number.isInteger(id)) {
    throw createError({
      statusCode: 400,

>       statusMessage: 'ID should an integer',

Reviewer: ```suggestion
      statusMessage: 'ID should be an integer',
```

})
```

### Error handling

If no errors are thrown a status code of `200 OK` will be returned. Any uncaught errors will return a `500 Internal Server Error` HTTP Error.

To return other error codes, throw an exception with `createError`

```ts [server/api/validation/[id].ts]
export default defineEventHandler((event) => {
  const id = parseInt(event.context.params.id) as number
  if (!Number.isInteger(id)) {
    throw createError({
      statusCode: 400,
      statusMessage: 'ID should an integer',
    })
  }
  return 'All good'
})
```

### Returning other status codes

> To return other status codes, modify `event.node.res.statusCode`.

Reviewer: In fact, to return 204 you can directly return null, or use the `sendNoContent` utility.

For here, I would recommend instead using the `setResponseStatus` utility.

```suggestion
To return other status codes, you can use the `setResponseStatus` utility.
```

> For example to return `204 No Content`

Reviewer: ```suggestion
For example to return `202 Accepted`
```

```ts [server/api/validation/[id].ts]
export default defineEventHandler((event) => {

>   event.node.res.statusCode = 204
>   return {}

Reviewer: ```suggestion
  setResponseStatus(event, 202)
```

> For example to return `204 No Content`
> 
> ```ts [server/api/validation/[id].ts]
> export default defineEventHandler((event) => {
>   event.node.res.statusCode = 204
>   return {}
> })
> ```

Reviewer: ```suggestion
For example, to return `202 Accepted`

```ts [server/api/validation/[id].ts]
export default defineEventHandler((event) => {
  setResponseStatus(event, 202)
})
```
```
